core/controllers/library_test.py

```diff
@@ -194,14 +197,148 @@ def test_library_handler_for_created_explorations(self):
+        self.assertDictContainsSubset({
+            'activity_summary_dicts_by_category': [],
+            'preferred_language_codes': ['en'],
+        }, response_dict)
+
+        # Load a demo.
+        exp_services.load_demo('0')
+        rating_services.assign_rating_to_exploration('user', '0', 2)
+        response_dict = self.get_json(feconf.LIBRARY_INDEX_DATA_URL)
+        self.assertEqual(
+            len(response_dict['activity_summary_dicts_by_category']), 1)
+        self.assertDictContainsSubset({
+            'preferred_language_codes': ['en'],
+        }, response_dict)
+        activity_summary_dicts_by_category = (
+            response_dict['activity_summary_dicts_by_category'][0])
+        self.assertDictContainsSubset({
+            'categories': [],
+            'header_i18n_id': (
+                feconf.LIBRARY_CATEGORY_TOP_RATED_EXPLORATIONS),
+            'has_full_results_page': True,
+            'full_results_url': feconf.LIBRARY_TOP_RATED_URL,
+            'protractor_id': 'top-rated',
+        }, activity_summary_dicts_by_category)
+        activity_summary_dicts = (
+            activity_summary_dicts_by_category['activity_summary_dicts'])
+        self.assertEqual(
+            len(activity_summary_dicts), 1)
+        self.assertDictContainsSubset({
+            'id': '0',
+            'category': 'Welcome',
+            'title': 'Welcome to Oppia!',
+            'language_code': 'en',
+            'objective': 'become familiar with Oppia\'s capabilities',
+            'status': rights_manager.ACTIVITY_STATUS_PUBLIC,
+        }, activity_summary_dicts[0])
+
+    def test_library_index_handler_for_featured(self):
+        """Test the handler for featured explorations."""
+        response_dict = self.get_json(feconf.LIBRARY_INDEX_DATA_URL)
+        self.assertDictContainsSubset({
+            'activity_summary_dicts_by_category': [],
+            'preferred_language_codes': ['en'],
+        }, response_dict)
+
+        # Load a demo.
+        exp_services.load_demo('0')
+        exploration_ref = activity_domain.ActivityReference(
+            constants.ACTIVITY_TYPE_EXPLORATION, '0')
+        activity_services.update_featured_activity_references([exploration_ref])
+
+        response_dict = self.get_json(feconf.LIBRARY_INDEX_DATA_URL)
+        self.assertEqual(
+            len(response_dict['activity_summary_dicts_by_category']), 1)
+        self.assertDictContainsSubset({
+            'preferred_language_codes': ['en'],
+        }, response_dict)
+        activity_summary_dicts_by_category = (
+            response_dict['activity_summary_dicts_by_category'][0])
+        self.assertDictContainsSubset({
+            'categories': [],
+            'header_i18n_id': (
+                feconf.LIBRARY_CATEGORY_FEATURED_ACTIVITIES),
+            'has_full_results_page': False,
+            'full_results_url': None,
+        }, activity_summary_dicts_by_category)
+        activity_summary_dicts = (
```

Comment:
Use some spacing between each stages of testing, i.e separate the assert statements, the request statements and other assignment statements. Looks kinda neat. :)

Reply:
fixed in the commit